### ISSUE TITLE
When will you release the codes and models?

### ISSUE THREAD

**Issue by Susan19900316** [JasonBoy1/TextZoom#1]

**Comment by JasonBoy1** [comment]
Maybe in 2 weeks..Sorry, busy on my work recently.

**Comment by williamking5** [comment]
> Maybe in 2 weeks..Sorry, busy on my work recently.

Maybe you can release the dataset firstly

**Comment by JasonBoy1** [comment]
> > Maybe in 2 weeks..Sorry, busy on my work recently.
> 
> Maybe you can release the dataset firstly

Sure.

**Comment by sdtblck** [comment]
Any updates on this?

**Comment by JasonBoy1** [comment]
> Any updates on this?

Dataset and annotation updated.😁

**Comment by sdtblck** [comment]
Amazing! Are there plans to release the model code, too?

**Comment by sdtblck** [comment]
Also, any chance for alternative download sources than Baidu? 
If you can send me the dataset I'd be happy to host it on my drive

**Comment by victorca25** [comment]
> Also, any chance for alternative download sources than Baidu?
+1 to that Baidu is basically impossible to use outside of China

**Comment by JasonBoy1** [comment]
> > Also, any chance for alternative download sources than Baidu?
> 
> +1 to that Baidu is basically impossible to use outside of China

Sure, I will upload them to Google Drive.

**Comment by sayanxtreme** [comment]
I noticed that the LR and HR pairs have same size. In page 6, you've mentioned creating a 2x (16 LR, 32 HR) train set. Since we cannot introduce interpolation degradation and both images of pair having same size, how do I create the set? Can you please release the 2x trainset?

**Comment by JasonBoy1** [comment]
> I noticed that the LR and HR pairs have same size. In page 6, you've mentioned creating a 2x (16 LR, 32 HR) train set. Since we cannot introduce interpolation degradation and both images of pair having same size, how do I create the set? Can you please release the 2x trainset?

This is because the original LR images are upsampled to the size of HR images. 'Upsample' is different from downsample degradation, the information is not reduced. The author of realSR and SRRAW also uses upsample+align crop+downsample to generate images. You can either downsample them to their original size or crop the text images from SRRAW and RealSR following my annotations. 

In my released dataset, you can directly resize the LR image to 16 pixels height and HR images to 32 pixels height. You will find that the accuracy of resized LR images and not resized LR images is approximate to each other. Because their original size is no bigger than 16 pixels. Upsample will not change much.

**Comment by sayanxtreme** [comment]
Got it. Thanks for the quick reply!

**Comment by pavlos13** [comment]
So, in order to create the 2x datasets, all the LR images and HR images should be resized to (64,16) and (128,32) respectively?

**Comment by JasonBoy1** [comment]
> So, in order to create the 2x datasets, all the LR images and HR images should be resized to (64,16) and (128,32) respectively?

Yes, the size of HR images is no large than 128X32, and the size of LR images is no large than 64X16. You will find that their accuracy changes little.

**Comment by pavlos13** [comment]
Great! Thank you

**Comment by pavlos13** [comment]
Another question.This also applies for testsets? i.e. When you evaluate on easy, medium and hard test sets, you take into account only the images with heights in [8, 16] and [16, 32] for LR and HR respectively?

**Comment by JasonBoy1** [comment]
> Another question.This also applies for testsets? i.e. When you evaluate on easy, medium and hard test sets, you take into account only the images with heights in [8, 16] and [16, 32] for LR and HR respectively?

Yes. All the selected LRs are in [8, 16] and all the HRs are in [16,32]. I resized all the LRs to 64*16 and all the HRs to 128*32.

**Comment by mosay95** [comment]
When will you release the code? I have waitting for half year since "TextSR" paper.

**Comment by pavlos13** [comment]
Hello, thanks for uploading the code! Is there any possibility to upload the pre-trained models?
Thank you!
